### PR TITLE
Fixes for O365DSC Support

### PR DESCRIPTION
**Pull request by NikCharlebois**
Need to validate tests against OA before merging

### CODE REVIEW

File: MSCloudLoginAssistant.psm1 [hunk 1]
```diff
@@ -36,32 +42,39 @@ function Test-MSCloudLogin
...
         }
-        "ExchangeOnline"
+        'ExchangeOnline'
         {
             $testCmdlet = "Get-Mailbox";
             $exceptionStringMFA = "AADSTS";
             $connectCmdlet = "Connect-EXOPSSession";
             $connectCmdletArgs = "-Credential `$o365Credential";
             $connectCmdletMfaRetryArgs = "-UserPrincipalName `$o365Credential.UserName";
+            $connectCmdletArgs = $connectCmdletMfaRetryArgs;
```
Comment (brianlala) [inline MSCloudLoginAssistant.psm1]: Duplicate?

File: MSCloudLoginAssistant.psm1 [hunk 2]
```diff
@@ -74,34 +87,31 @@ function Test-MSCloudLogin
...
-        "PnP"
+        'PnP'
         {
-            $global:spoAdminUrl = Get-SPOAdminUrl;
             $testCmdlet = "Get-PnPSite";
             $exceptionStringMFA = "sign-in name or password does not match one in the Microsoft account system";
             $connectCmdlet = "Connect-PnPOnline";
-            $connectCmdletArgs = "-TenantAdminUrl $global:spoAdminUrl -Url $(($global:spoAdminUrl).Replace('-admin','')) -Credentials `$o365Credential";
-            $connectCmdletMfaRetryArgs = $connectCmdletArgs.Replace("-Credentials `$o365Credential","-UseWebLogin");
+            $connectCmdletArgs = "-Url $ConnectionUrl -Credentials `$O365Credential";
```
Comment (brianlala) [inline MSCloudLoginAssistant.psm1:104]: Why are we removing the MFA retry logic here?

File: MSCloudLoginAssistant.psm1 [hunk 3]
```diff
@@ -204,26 +219,10 @@ function Test-MSCloudLogin
     {
         if (Get-Variable -Name $variablePrefix"LoginSucceeded" -ValueOnly -Scope "Global")
         {
-            Write-Host -ForegroundColor Green " - Successfully logged in to $Platform."
+            Write-Verbose " - Successfully logged in to $Platform."
```
Comment (brianlala) [inline MSCloudLoginAssistant.psm1]: This is OK but let's use -Message for consistency :)

File: MSCloudLoginAssistant.psm1 [hunk 4]
```diff
@@ -36,32 +42,39 @@ function Test-MSCloudLogin
...
-        "AzureAD"
+        'AzureAD'
         {
             $testCmdlet = "Get-AzureADUser";
             $exceptionStringMFA = "AADSTS";
             $connectCmdlet = "Connect-AzureAD";
-            $connectCmdletArgs = "-Credential `$o365Credential";
-            $connectCmdletMfaRetryArgs = "-AccountId `$o365Credential.UserName";
+            $connectCmdletArgs = "-Credential `$O365Credential";
+            $connectCmdletMfaRetryArgs = ""
```
Comment (brianlala) [inline MSCloudLoginAssistant.psm1]: Why did we change this to blank?

File: MSCloudLoginAssistant.psm1 [hunk 5]
```diff
@@ -36,32 +42,39 @@ function Test-MSCloudLogin
...
+            {
+                $global:spoAdminUrl = $ConnectionUrl
+            }
             $testCmdlet = "Get-SPOSite";
             $exceptionStringMFA = "sign-in name or password does not match one in the Microsoft account system";
             $connectCmdlet = "Connect-SPOService";
-            $connectCmdletArgs = "-Url $global:spoAdminUrl -Credential `$o365Credential";
-            $connectCmdletMfaRetryArgs = $connectCmdletArgs.Replace("-Credential `$o365Credential","");
+            $connectCmdletArgs = "-Url $global:spoAdminUrl -Credential `$O365Credential";
             $variablePrefix = "spo"
```
Comment (brianlala) [inline MSCloudLoginAssistant.psm1:69]: $connectCmdletMfaRetryArgs is used when we catch a login error related to MFA, why did we remove this?

**Review by brianlala — CHANGES_REQUESTED** [review]
Only real concern is about changes to $connectCmdletMfaRetryArgs.

File: MSCloudLoginAssistant.psm1 (hunk 1 above)
Comment (NikCharlebois) [reply MSCloudLoginAssistant.psm1]: Probably merge conflict. EXO and SC connection will still be handled outside of MSCloudLogin for the moment due to their level of complexity

File: MSCloudLoginAssistant.psm1 (hunk 2 above)
Comment (NikCharlebois) [reply MSCloudLoginAssistant.psm1:104]: The real retry for MFA has no parameters. I will put $connectCmdletMfaRetryArgs = "" just for clarity's sake

File: MSCloudLoginAssistant.psm1 (hunk 4 above)
Comment (NikCharlebois) [reply MSCloudLoginAssistant.psm1]: Added back

File: MSCloudLoginAssistant.psm1 (hunk 5 above)
Comment (NikCharlebois) [reply MSCloudLoginAssistant.psm1:69]: Again, they are empty, adding $connectCmdletMfaRetryArgs = "" back

**Review by brianlala — APPROVED** [review]
Sweet

**Pull request merged**